rcl/include/rcl/arguments.h

```diff
@@ -39,15 +39,19 @@ typedef struct rcl_arguments_t
 #define RCL_ROS_ARGS_EXPLICIT_END_TOKEN "--"
 #define RCL_PARAM_FLAG "--param"
 #define RCL_SHORT_PARAM_FLAG "-p"
+#define RCL_PARAM_FILE_FLAG "--params-file"
+#define RCL_SHORT_PARAM_FILE_FLAG "-f"
 #define RCL_REMAP_FLAG "--remap"
 #define RCL_SHORT_REMAP_FLAG "-r"
+#define RCL_LOG_LEVEL_FLAG "--log-level"
+#define RCL_SHORT_LOG_LEVEL_FLAG "-l"
+#define RCL_EXTERNAL_LOG_CONFIG_FLAG "--log-config-file"
+#define RCL_SHORT_EXTERNAL_LOG_CONFIG_FLAG "-c"
+// To be prefixed with --enable- or --disable-
+#define RCL_LOG_STDOUT_FLAG_SUFFIX "stdout-logs"
+#define RCL_LOG_ROSOUT_FLAG_SUFFIX "rosout-logs"
+#define RCL_LOG_EXT_LIB_FLAG_SUFFIX "external-lib-logs"
 
-#define RCL_LOG_LEVEL_ARG_RULE "__log_level:="
```

Comment:
There're two related questions that need an answer here. Thanks for bringing it up again.

First, whether we want to drop the old rules or not. If we don't, that'd preclude supporting multiple values per flag (e.g. multiple remapping rules per `-r` flag). That's not supported nor in the design document, but I've seen interest expressed elsewhere.  

Assuming that the answer to the first question is `yes`, how do we want to drop them? This is simply removing them, but we may want to go through a deprecation cycle to allow the community to settle. I'd like to hear what @dirk-thomas and @wjwwood think about this.

Reply:
I had thought we'd settled on keeping the old rules so there is less disruption, but I may have imagined it.

I'm in favor of dropping them though with a deprecation cycle.

Reply:
Another detail I had not thought of is how far should we go back. This PR only or back to the start, with known limitations?

Reply:
What start are you referring to?
So, I think that `--ros-args`, `-r`, `-p` should we always be enforced, without backwards compatibility. If not, it will be quite confusing.

About changing some of the `__*:=*` options for flags, I think we can support both for a time, first just deprecating `__*:=*` and the deleting them.

But as we will broke the API, by adding `--ros-args` and forcing `-r` `-p`, I think is also ok to just delete the old `__*:=*` options.

Reply:
Right, that's sort of my point. If we strictly abide to a deprecation cycle we need to support arguments outside `--ros-args`, the ones that used to be there in Dashing. Otherwise, it's like half breaking compatibility.

Reply:
I have to agree with @jacobperron that we should keep the old flags for now.  While it is somewhat annoying, breaking everything in one release will lead to quite some fallout for downstream users (and it isn't just code; they'll also have to retrain their fingers to do something different).  So I'd be for deprecating the old `__*:=*` in Eloquent, then removing them in F-turtle.

Which brings up a point about how we go about deprecating them.  Obviously we can put it in documentation, but should we also print a console message if the old flags are used?

Reply:
Alright, per offline discussion, we're going to re-instantiate support for old arguments and mark them as deprecated with a warning. I'll do so in another PR. Since it's orthogonal to this PR, we can defer merging till then.

Reply:
I take that back. It's much easier to do this in a follow-up PR, which I've written already. So if you don't mind, let's get this in and review the next PR right after.

Reply:
@jacobperron @ivanpauno ^

Reply:
Sorry for the delay. A follow-up PR is okay with me. I'll take another look at this one.